src/java_bytecode/java_bytecode_language.cpp

```diff
@@ -408,6 +409,30 @@ bool java_bytecode_languaget::convert_single_method(
       string_preprocess.code_for_function(symbol, symbol_table);
     INVARIANT(
       generated_code.is_not_nil(), "Couldn't retrieve code for string method");
+    // String solver can make calls to functions that haven't yet been seen.
+    // Add these to the needed_lazy_methods collection
+    if(needed_lazy_methods)
+    {
+      for(const_depth_iteratort it = generated_code.depth_cbegin();
+          it != generated_code.depth_cend();
+          ++it)
+      {
+        if(it->id() == ID_code)
+        {
+          const auto fn_call = expr_try_dynamic_cast<code_function_callt>(
+            static_cast<const exprt &>(*it));
```

Comment:
why do you need this static_cast?

Reply:
Because there's (rightly) no `to_expr` helper function.

Reply:
but isn't `*it` already an `exprt`?

Reply:
it's not (@LAJW could confirm). CLion agrees with me and says `casting expression to const exprt & is redundant`

Reply:
Oops, you're right, @NathanJPhillips this cast can go